CAM-14302: chore(engine): bump groovy to 2.5.16

koevskinikola opened the pull request:
related to CAM-14302

tmetzke commented on internal-dependencies/pom.xml:
```diff
@@ -138,11 +138,44 @@
       </dependency>
+
+      <!-- groovy script engine dependencies -->
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-jsr223</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-json</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
       <dependency>
         <groupId>org.codehaus.groovy</groupId>
-        <artifactId>groovy-all</artifactId>
+        <artifactId>groovy-xml</artifactId>
         <version>${version.groovy}</version>
       </dependency>
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-templates</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-dateutil</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-datetime</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
+
```
You can import the Groovy BOM so you don't need to track all versions separately.

```suggestion
      <!-- groovy script engine dependencies -->
      <dependency>
        <groupId>org.codehaus.groovy</groupId>
        <artifactId>groovy-all</artifactId>
        <version>${version.groovy}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
```

koevskinikola replied:
I actually tried this first, but for some reason, Maven didn't make the Groovy dependencies available to the child poms. I was getting a "missing version" error. I'll try again, and ping you if I run into some problems.

koevskinikola replied:
The solution was to use `groovy-bom` instead of `groovy-all` since the latter is not a BOM.

tmetzke replied:
Sorry for the mix-up 😄👍

tmetzke reviewed (changes requested):
The update looks good overall. 👍 

Some things I think can be improved or discussed:

* We can import the `groovy-all` BOM in the `internal-dependencies` to avoid tracking all artifacts separately. 
* We could also remove Groovy version management from internal dependencies completely and import the Groovy BOM where needed. This would remove the necessity fo the overrides in `engine-rest`. However, we'll need a lot more imports in other modules then. I am undecided, to be honest.
* Run seems to use the wrong version for Groovy (`3.0.10` as provided by Spring Boot)

koevskinikola commented on distro/run/assembly/pom.xml:
```diff
@@ -91,8 +91,27 @@
       <groupId>org.codehaus.groovy</groupId>
-      <artifactId>groovy-all</artifactId>
-      <version>${version.groovy}</version>
+      <artifactId>groovy-jsr223</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-json</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-xml</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-datetime</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-dateutil</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-templates</artifactId>
```
I missed that, nice catch. I'll work on it.

tmetzke reviewed (approved):
That looks great now 👍 
Let's wait for CI feedback but I don't expect any hiccups 🙂

koevskinikola commented:
@tmetzke there was a hiccup :) When I added the `groovy-bom` to the Camunda Run Root pom, I forgot that it would override the Groovy version used in `rest-assured` inside the Run QA module. I moved the `groovy-bom` into the Run `assembly` module.

Things look good now. :smile:

tmetzke reviewed (approved):
Good thing we have that CI running for us 😁 
Looks good 👍 
Did you manually check Run artifacts that result from the build to contain the right Groovy version by any chance? 🙂

koevskinikola commented:
> Good thing we have that CI running for us grin Looks good +1 Did you manually check Run artifacts that result from the build to contain the right Groovy version by any chance? slightly_smiling_face

I haven't been able to. I'm getting some strange compilation failures in the Run example module locally. The issue is not present in the CI. Can you try it on your side?

tmetzke commented:
> > Good thing we have that CI running for us grin Looks good +1 Did you manually check Run artifacts that result from the build to contain the right Groovy version by any chance? slightly_smiling_face
> 
> I haven't been able to. I'm getting some strange compilation failures in the Run example module locally. The issue is not present in the CI. Can you try it on your side?

I will give it a try and update here 👍

koevskinikola commented:
I managed to build it. All of the 2.5.16 `groovy-*`  artifacts that I declare are in the `userlib/` directory.

There was a local build issue with the Maven reactor. I had to build everything using `mvn clean install -pl distro/run -am -amd -DskipTests`.

The `-am` builds all the required parent modules, and the `-amd` builds all the child modules. This worked out pretty nice :)

tmetzke commented:
> The `-am` builds all the required parent modules, and the `-amd` builds all the child modules. This worked out pretty nice :)

That's a neat trick, I discovered that last week as well "by accident" 😄
